.github/workflows/validate-pr.yml

```diff
@@ -0,0 +1,61 @@
+# Validates the format of a PR title and body. Because the title will become the default commit message on squash-merge,
+# it should follow the conventional commit format and be 70 characters long or fewer. Because the body will later be
+# used in creating docs-site release notes, it must have a summary paragraph (70+ characters) followed by `\n---`.
+
+name: "Validate PR"
+on:
+  pull_request:
+    types:
+      - opened
+      - edited
+      - synchronize
+
+jobs:
+  validate-pr-format:
+    name: Validate PR Format
+    runs-on: ubuntu-latest
+    steps:
+      - name: Parse PR Contents
+        id: parse-pr-contents
+        uses: actions/github-script@v6
+        with:
+          script: |
+            // Helpers for logging
+            const head = (text) => { core.info(`\u001b[36;1m${text}`) } // cyan bold
```

Comment:
This does not need to be fixed before merging, but IMO We should avoid writing long scripts in this way.  We lose some of our linting and IDE magic like autocomplete and syntax formatting that help detect issues when its implemented like this.

Reply:
True. As the script got long I considered pulling the code in from a separate file. The benefit of no checkout action is speed, and I figured we'd end up glad for a faster feedback cycle.